### ISSUE TITLE
Issue with resnet50, resnet101, and resnet152

### ISSUE THREAD

**Issue by ErrorGz** [dotnet/TorchSharp#1001]
Dear contributors,

I am writing to report an issue and request your help and guidance with a training model program for image classification that I have been developing using TorchSharp. The repository is located at https://github.com/ErrorGz/ImageProcess/tree/master/ImageTrain, and I am pleased to report that the loss and accuracy are converging properly.
In the chart, the black line represents the first training session, while the blue line represents the continuation of training after loading the model from the first session.

However, I have encountered an error when attempting to use the resnet50, resnet101, and resnet152 models. Specifically, the program runs smoothly with torchvision.models.resnet18 and resnet34, but when I attempt to use the aforementioned models, I receive the following error:

_System.Runtime.InteropServices.ExternalException: "Given groups=1, weight of size [64, 64, 3, 3], expected input[16, 256, 104, 104] to have 64 channels, but got 256 channels instead
Exception raised from check_shape_forward at C:\actions-runner_work\pytorch\pytorch\builder\windows\pytorch\aten\src\ATen\native\Convolution.cpp:565 (most recentcall first)"_

I would greatly appreciate any suggestions or guidance that you may have to resolve this issue. Thank you for your contributions and support.

Best regards,

![2023-05-19_142625](https://github.com/dotnet/TorchSharp/assets/32073343/b5552951-12a8-4b22-8030-7383ab052c7f)

![2023-05-19_211341](https://github.com/dotnet/TorchSharp/assets/32073343/471c30ae-7e00-4911-987d-cf4f60c95244)

**Comment by NiklasGustafsson** [comment]
Thanks for the issue report!

The ResNet models are defined in:

https://github.com/dotnet/TorchSharp/blob/main/src/TorchVision/models/ResNet.cs

based on the implementation at:

https://github.com/pytorch/vision/blob/main/torchvision/models/resnet.py

What the three models you have trouble with have in common is that the repeated block is called `Bottleneck`, whereas the smaller models are based on a block called `BasicBlock`. I will take a look at the details, but it is possible that there's a bug in the implementation of `Bottleneck` that has slipped through.

Feel free to take a look yourself, too.

**Comment by NiklasGustafsson** [comment]
Looking at the code, I'm pretty sure the implementation of `Bottleneck` is at the center of this.

**Comment by NiklasGustafsson** [comment]
@ErrorGz -- if you want to take a look at an attempt at cleaning up the code, here's a draft:

https://github.com/NiklasGustafsson/TorchSharp/blob/bugs/src/TorchVision/models/ResNet.cs

You should be able to copy the file and incorporate it into your program and see if it works better for the large architectures.

**Comment by NiklasGustafsson** [comment]
@ErrorGz -- did you have a chance to look at the new versions?
